R/class-plotdata-line.R

```diff
@@ -52,35 +55,51 @@ newLinePD <- function(.dt = data.table::data.table(),
   attr <- attributes(.pd)
 
   x <- veupathUtils::toColNameOrNull(attr$xAxisVariable)
+  xType <- attr$xAxisVariable$dataType
   y <- veupathUtils::toColNameOrNull(attr$yAxisVariable)
   group <- veupathUtils::toColNameOrNull(attr$overlayVariable)
   panel <- findPanelColName(attr$facetVariable1, attr$facetVariable2)
 
+  # think we need to take viewport as input, even if we dont want semantic zoom
+  # for consistent bins across the annotated range, we need a consistent range/ bin start
+  if (is.null(viewport)) {
+    viewport <- findViewport(.pd[[x]], xType)
+    veupathUtils::logWithTime('Determined default viewport.', verbose)
+  } else {
+    viewport <- validateViewport(viewport, xType)
+  }
+  attr$viewport <- lapply(viewport, as.character)
+  attr$viewport <- lapply(attr$viewport, jsonlite::unbox)
+
+  # if no binWidth provided, should we find one or assume a binWidth of 0?
+  # does a binWidth of 0 really make sense, or do we need to make all of the binning stuff optional?
+  # ultimately the question is how to make binning optionaland whats the default behavior?
```

Comment:
i think we decided to have default no binning. But i guess what if the user turns on binning and sets the bin width to 0. Should that be different than just returning all the points? I suppose on one hand if no binning = binwidth 0 there's a possibility to have a more consistent response, but if they're different it's more consistent with how the viz might look from the user's perspective, so is a little more clear in that way. 

Actually if i was using plot.data outside of the eda and i just wanted to return line plot data, i would not expect no binning to be the same as binWidth 0. I'd expect to just get the series back. Not sure if that's helpful :)

Reply:
yea i didnt want to have to do it but im thinking we need binning to be properly optional..